### PR TITLE
Add status

### PR DESCRIPTION
**Pull request by eliegaboriau**
This PR enables status to be used.

You can create, update and destroy status the same way as areas for example, and you can name your status the way you want.
You can then add a status to an initiative when updating it, and this status will be displayed with the tags on the initiative card and on the initiative page.
You can also filter initiatives by status on the initiatives page.

How to test : 

- Go to admin settings
- GO to statuses
- Create a new status
- Go to admin initiatives
- Update a published initiative, and add your status
- Go to the public page of this initiative
- See your status in the tags
- Go to initiatives
- See your status in the initiative card
- Play with status filters to see it works.

For dev reviewers : 
Two erb files could have been defaced but 
- for the form, i wasn't able to find a good selector to insert into
- for the tags, it wasn't detected when calling it from the cell

### CODE REVIEW

**Review by luciegrau — COMMENTED** [review]
Hi Elie, thank you for your PR that works fine for me. I only have a feedback that concerns the behavior of the filters on the initiative index page. Indeed, when I click on one of the filters on an initiative card, I expect to access a filtered view of the initiatives according to the filter I have chosen. Instead, I have no filtered view and all filters remain checked. This bug seems to concern Decidim from version 0.27 since I could reproduce it on try.decidim.org (v0.27) and nightly.decidim.org (v0.28).

**Comment by eliegaboriau** [comment]
@luciegrau It should be okay for the filters now !

**Comment by luciegrau** [comment]
Hi Elie, thank you for fix ! I just have two feedbacks after I tried locally : 

- The status filters are not displayed in the BO, it means as an admin you can't filter initiatives by their status in the BO ; 

Capture d’écran 2023-02-21 à 16 14 39

- Filters are working well in the FO but the checkboxes of the scopes do not appear as checked when a user checks them.

**Comment by eliegaboriau** [comment]
Hello @luciegrau 
About the admin filters, it should now be okay
Abour the scope filters, the problem is when there are subscopes. Knowing that we won't use nether scopes nor subscopes, we may not fix it.

**Review by luciegrau — APPROVED** [review]
Hi Elie, it works great thank you 💪

File: app/controllers/decidim/admin/statuses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          on(:has_spaces) do
+            flash[:alert] = I18n.t("statuses.destroy.has_spaces", scope: "decidim.admin")
+            redirect_to statuses_path
+          end
+        end
+      end
+
+      private
+
+      def organization_statuses
```
Comment (Quentinchampenois) [inline app/controllers/decidim/admin/statuses_controller.rb:76]: `organization_statuses` can be memoized 

```suggestion
      def organization_statuses
          @organization_statuses ||= current_organization.statuses
      end
```

**Review by Quentinchampenois — APPROVED** [review]
Nice work ! 

I just have a little improvement in statuses controller but nothing significative

I did not execute module locally since we already have done the product review

**Pull request merged**